configstack/running_module.go

```diff
@@ -167,17 +183,37 @@ func runModules(modules map[string]*runningModule) error {
 // occurred
 func collectErrors(modules map[string]*runningModule) error {
 	errs := []error{}
+	detailedErrs := []error{}
+
 	for _, module := range modules {
 		if module.Err != nil {
 			errs = append(errs, module.Err)
+			detailedErrs = append(detailedErrs, generateDetailedErrorMessage(module))
 		}
 	}
 
 	if len(errs) == 0 {
 		return nil
+	}
+
+	return errors.WithStackTrace(MultiError{Errors: errs, DetailedErrors: detailedErrs})
+}
+
+// generateDetailedErrorMessage extracts the clean stderr from a module and formats it for printing
+func generateDetailedErrorMessage(module *runningModule) error {
+	// remove the auto-init pollution from the error stream
+	cleanErrorOutput := strings.Replace(module.ErrStream.String(), module.Module.TerragruntOptions.InitStream.String(), "", -1)
```

Comment:
Hm, string replacement feels... A bit hacky. Is there any way to have the AutoInit write to `stdout` / `stderr`, but to _not_ write to `module.ErrStream`? That way, you could use `module.ErrStream` directly, without having to clean anything out...